core/src/main/java/org/apache/calcite/plan/volcano/RelSet.java

```diff
@@ -73,7 +74,14 @@
    * set.
    */
   RelSet equivalentSet;
-  RelNode rel;
+
+  /**
+   * The first RelNode added to the set or the RelNode with highest confidence
+   * level of estimated statistics.
+   * The logical properties of the RelSet, including row count, uniqueness, etc,
+   * are determined by this RelNode.
+   */
+  RelNode originalRel;
```

Comment:
The name is a little bit misleading. Before this patch, it is indeed original rel, but after this patch, it isn't original rel anymore. We could just call it as it is.

Reply:
Yep, I find it awkward too. Do you have any suggestions?

Reply:
How about continue using `rel`?